mycroft/skills/phillips_hue/__init__.py

```diff
@@ -0,0 +1,283 @@
+
+        decrease_brightness_intent = IntentBuilder("DecreaseBrightnessIntent")\
+            .require("DecreaseBrightnessKeyword").build()
+        self.register_intent(decrease_brightness_intent,
+                             self.handle_intent)
+
+        increase_brightness_intent = IntentBuilder("IncreaseBrightnessIntent")\
+            .require("IncreaseBrightnessKeyword").build()
+        self.register_intent(increase_brightness_intent,
+                             self.handle_intent)
+
+        decrease_color_temperature_intent =\
+            IntentBuilder("DecreaseColorTemperatureIntent")\
+            .require("DecreaseColorTemperatureKeyword").build()
+        self.register_intent(decrease_color_temperature_intent,
+                             self.handle_intent)
+
+        increase_color_temperature_intent =\
+            IntentBuilder("IncreaseColorTemperatureIntent")\
+            .require("IncreaseColorTemperatureKeyword").build()
+        self.register_intent(increase_color_temperature_intent,
+                             self.handle_intent)
+
+        connect_lights_intent = \
+            IntentBuilder("ConnectLightsIntent") \
+            .require("ConnectLightsKeyword").build()
+        self.register_intent(connect_lights_intent,
+                             self.handle_intent)
+
+    def handle_intent(self, message):
+        if message.message_type == 'ConnectLightsIntent'\
+                or self.connected or self._connect_to_bridge():
+            try:
+                if message.message_type == "TurnOffIntent":
+                    self.handle_turn_off_intent(message)
+                elif message.message_type == "TurnOnIntent":
+                    self.handle_turn_on_intent(message)
+                elif message.message_type == "ActivateSceneIntent":
+                    self.handle_activate_scene_intent(message)
+                elif message.message_type == "DecreaseBrightnessIntent":
+                    self.handle_decrease_brightness_intent(message)
+                elif message.message_type == "IncreaseBrightnessIntent":
+                    self.handle_increase_brightness_intent(message)
+                elif message.message_type == "DecreaseColorTemperatureIntent":
+                    self.handle_decrease_color_temperature_intent(message)
+                elif message.message_type == "IncreaseColorTemperatureIntent":
+                    self.handle_increase_color_temperature_intent(message)
+                elif message.message_type == "ConnectLightsIntent":
+                    self.handle_connect_lights_intent(message)
+                else:
+                    raise Exception('No matching intent handler')
+            except Exception as e:
+                if isinstance(e, PhueRequestTimeout):
+                    self.speak_dialog('unable.to.perform.action')
+                elif 'No route to host' in e.args:
+                    if not self.config.get('ip', '') == '':
```

Comment:
self.ip can be set via the value in the config, or by trying to discover the bridge on the network. Here, we must check if it was supplied in the config, because we only want to speak the no.route dialog if the user supplied the ip. This is because with a user supplied ip, we will never try to change it, only alert them if there is an issue using it.

I should probably update the ip if we get here and it's not user supplied though. I think I was a little too focused on handling the case where the ip was user supplied, and not enough on resolving the issue otherwise.